### ISSUE TITLE
Lint/ToEnumArguments doesn't support anonymous argument forwarding

### ISSUE THREAD

**Issue by inkstak** [rubocop/rubocop#12143]
## Expected behavior

Since anonymous arguments forwarding is enforced by `Style/ArgumentsForwarding`, 
the code below should also be validated by `Lint/ToEnumArguments`

```ruby
def do_something(**)
  return to_enum(:do_something, **) unless block_given?

  do_something_else
end
```

## Actual behavior

The code return the following issue:

```
file.rb:10:20: W: Lint/ToEnumArguments: Ensure you correctly provided all the arguments.
            return to_enum(:do_something, **) unless block_given?
                   ^^^^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

Create a file with example above, then run `bundle exec rubocop --only Lint/ToEnumArguments example.rb`

## RuboCop version

```
$ [bundle exec] rubocop -V
1.56.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.1) +server [arm64-darwin22]
  - rubocop-factory_bot 2.23.1
  - rubocop-performance 1.19.0
  - rubocop-rails 2.20.2
  - rubocop-rspec 2.23.2
```
